### DISCUSSION TITLE
Operator Error for data type mismatch

### DISCUSSION THREAD

**Discussion by theShreyans** [supabase/supabase#9354] [Questions]
I don't know if this is a bug but I'm getting this error every now and then.
Error: 
**operator does not exist: uuid = text**

I'm using flutter sdk and my query is as follow: 
```dart
   final user = await Supabase.instance.client
        .from('users')
        .select('*')
        .match({"userUid": userUid}).execute();
```
In the users table whenever someone signs up, their generated id is stored under userUid colum that is of type uuid.

In the above query I'm getting the currently logged in users id from the session and passing it in the query to fetch their data. However the query would work sometime and randomly I will get the above mentioned error of **operator does not exist**. Has anyone else faced this problem? Also does anyone knows what is causing this? As I imagine that queries from client side would be read as string or int datatype depending on the column datatype to filter the data.

**Comment by GaryAustin1** [comment]
If for some reason userUid is ever empty string ('') it would generate that error.

**Reply by theShreyans** [reply]
Thank you for the response. I tried your solution and it seems like that, that's not the issue as I experimented by explicitly writing the userId directly in the query and the results were the same. Also, I experimented with a null string, resulting in a different error. As of now, it seems like the error is gone after I updated the previous version of the superbase_flutter package to a newer prerelease one.
